test/extensions/filters/common/ext_authz/ext_authz_http_impl_test.cc

```diff
@@ -318,11 +320,65 @@ TEST_F(ExtAuthzHttpClientTest, AuthorizationOkWithAddedAuthzHeaders) {
   EXPECT_CALL(*child_span,
               setTag(Eq(Tracing::Tags::get().UpstreamCluster), Eq(config_->cluster())));
   EXPECT_CALL(*child_span, injectContext(_));
-  // Expect that header1 will be added and header2 correctly overwritten.
-  EXPECT_CALL(async_client_, send_(AllOf(ContainsPairAsHeader(config_->headersToAdd().front()),
-                                         ContainsPairAsHeader(config_->headersToAdd().back())),
-                                   _, _));
-  client_->check(request_callbacks_, request, active_span_);
+  // Expect that header1 will be added and header2 correctly overwritten. Due to this behavior, the
+  // append property of header value option should always be false.
+  const std::pair<const Http::LowerCaseString, const std::string> header1{"x-authz-header1",
+                                                                          "value"};
+  const std::pair<const Http::LowerCaseString, const std::string> header2{"x-authz-header2",
+                                                                          "value"};
+  EXPECT_CALL(async_client_,
+              send_(AllOf(ContainsPairAsHeader(header1), ContainsPairAsHeader(header2)), _, _));
+  client_->check(request_callbacks_, request, active_span_, stream_info_);
+
+  EXPECT_CALL(request_callbacks_,
+              onComplete_(WhenDynamicCastTo<ResponsePtr&>(AuthzOkResponse(authz_response))));
+  EXPECT_CALL(*child_span, setTag(Eq("ext_authz_status"), Eq("ext_authz_ok")));
+  EXPECT_CALL(*child_span, setTag(Eq("ext_authz_http_status"), Eq("OK")));
+  EXPECT_CALL(*child_span, finishSpan());
+  client_->onSuccess(std::move(check_response));
+}
+
+// Verify client response headers when authorization_headers_to_add is configured with value from
+// stream info.
+TEST_F(ExtAuthzHttpClientTest, AuthorizationOkWithAddedAuthzHeadersFromStreamInfo) {
+  const std::string yaml = R"EOF(
+  http_service:
+    server_uri:
+      uri: "ext_authz:9000"
+      cluster: "ext_authz"
+      timeout: 0.25s
+    authorization_request:
+      headers_to_add:
+      - key: "x-authz-header1"
+        value: "%REQ(x-request-id)%"
+  failure_mode_allow: true
+  )EOF";
+
+  initialize(yaml);
+
+  Tracing::MockSpan* child_span{new Tracing::MockSpan()};
+  const auto expected_headers = TestCommon::makeHeaderValueOption({{":status", "200", false}});
+  const auto authz_response = TestCommon::makeAuthzResponse(CheckStatus::OK);
+  auto check_response = TestCommon::makeMessageResponse(expected_headers);
+
+  EXPECT_CALL(active_span_, spawnChild_(_, config_->tracingName(), _)).WillOnce(Return(child_span));
+  EXPECT_CALL(*child_span,
+              setTag(Eq(Tracing::Tags::get().UpstreamCluster), Eq(config_->cluster())));
+  EXPECT_CALL(*child_span, injectContext(_));
+
+  const std::pair<const Http::LowerCaseString, const std::string> headers_to_add{"x-authz-header1",
```

Comment:
name of this variable is a bit odd, i would just call this something like `expected_authz_header` or something